lighthouse-core/lib/network-recorder.js

```diff
@@ -205,6 +207,20 @@ class NetworkRecorder extends EventEmitter {
    * @return {NetworkRequest|undefined}
    */
   _findRealRequestAndSetSession(requestId, sessionId) {
+    // The very first sessionId processed is always the main sessionId. In all but DevTools,
+    // this sessionId is undefined. However, in DevTools the main Lighthouse protocol connection
+    // does send events with sessionId set to a string, because of how DevTools routes the protocol
+    // to Lighthouse.
+    // Many places in Lighthouse use `record.sessionId === undefined` to mean that the session is not
+    // an OOPIF. To maintain this property, we intercept sessionId here and set it to undefined if
+    // it matches the first value seen.
+    if (this._mainSessionId === null) {
```

Comment:
I wish the roles of `undefined` and `null` are reversed for `_mainSessionId`, but doing so would probably require changes to the protocol and/or large changes to how we handle `sessionId` downstream. Sounds like a lot of work for a minor annoyance, so this LGTM.

Reply:
we'll need to take another look at the entire use of `sessionId` when we revisit FR OOPIF handling which is going to have to be more complex anyway. I'm also fine with a temporary bandaid here